Post about Sunny tech (#56)
lunatech-labs/lunatech-blog: posts/2022-07-02-sunny-tech-en.adoc

= Sunny Tech 2022
VincentBrule
v1.0, 2022-07-02
:title: Sunny Tech 2022
:imagesdir: ../media/2022-07-02-sunny-tech
:lang: en
:tags: [conference, sunny tech, montpellier]

== Sunny Tech ?
https://sunny-tech.io/[Sunny Tech] took place from June 30 to July 1, under the sun of Montpellier.

This conference covers a multitude of areas:

. Cloud & Devops
. Methodologies
. Backend
. Data
. Web & Mobile & UI
. Architecture
. Atypical

The organizers have prepared a _great schedule for us_.
Thanks to the diversity of the topics presented, there was really something for every taste and profile.

> If you are interested, https://sunny-tech.io/schedule/[the program is available on the site].

Reviewer: Feels a little like switching to present tense here - perhaps...

``suggestion
If you are interested, https://sunny-tech.io/schedule/[the program is still available on the site].
```

The presentations had different formats.
First, the classic 50-minute presentations.
Then, workshops to learn by doing of varying length.

> And finally, short talk of 20 minutes.

Reviewer: ```suggestion
And finally, short talks of 20 minutes.
```

=== The organization

> Before presenting you my favorite discoveries of *Sunny Tech*, I would like to thank all the volunteers who made this event possible after several years of absence due to Covid.

Reviewer: You could go with

```suggestion
Before presenting to you my favourite discoveries of *Sunny Tech*, I would like to thank all the volunteers who made this event possible after several years of absence due to Covid.
```

or 

```suggestion
Before talking about my favourite discoveries of *Sunny Tech*, I would like to thank all the volunteers who made this event possible after several years of absence due to Covid.
```

Author: I like the second one thanks 👍

You don't realize it as a participant, but a titanic organization is necessary to set up a conference of this quality!

.Many thanks to the organizers (image taken by @ponceto91)
image::organisateur.jpeg[Many thanks to the organizers (image taken by @ponceto91), width = 700]

I would also like to highlight all the actions for the environment that have been implemented, such as:

. the possibility of not taking goodies and in exchange give the amount of these objects to an https://twitter.com/SunnyTech_MTP/status/1541701793322831873[association]
. a special booth for a local association, this year we were able to meet https://labelbleu.assoconnect.com/collect/description/233850-v-bulletin-d-adhesion-a-l-association-labelbleu-2022[LabelBleu]
. a 0-waste caterer who uses only reusable containers such as glass jars (and the food was very good, thanks to https://instant-bocal.com/[Instant bocal])

.Yum !
image::food.jpeg[Glass jars from Sunny Tech's caterer, width = 800]

> I will now present you my favorites of this conference!

Reviewer: see above

After many training cycles, the two modules will improve and become more and more efficient.

The first trick to detect false images generated by this model comes from the *image bank used*.
Indeed, the faces coming from this group of photos have a particularity: _the eyes are always placed in the same place_.
Therefore, the AI will try to create images as close as possible to the real ones and will therefore also place the eyes in the same place.
The most naive solution is to look at the position of the eyes on the image and if it matches the known position, the image is false.
But this solution is not very robust to changes in the pictures, so the presenter developed a more advanced method.

To do this, it has set up a service that retrieves and stores images from the site_thispersondoesnotexist_.
It has made several optimizations in order not to store the raw images on the server, as this would take up too much disk space.
Then, when you want to detect a fake face, the tool will compare with its database and according to a score, send back yes or no.
If you want to test, Mathis hosts a https://fada.h25.io/[website] allowing to detect fake images directly from the browser.

> Moreover, this web site displays interesting information such as look-alikes, when the fake image was generated, or the eyes technique explained before.

Reviewer: ```suggestion
Moreover, this website displays interesting information such as look-alikes, when the fake image was generated, or the eyes technique explained before.
```

> This talk is really well popularized and we are directly captured by the subject thanks to a very good speaker!

Reviewer: i'm not 100% clear what you're trying to say here - do you mean the talk was about a popular subject? Or that it was clearly explained? Sorry, i'm not quite getting this :(

=== "Il était une fois... les navigateurs" (Once upon a time... the browsers)

*Noël Macé* and *Pierre Tibulle* presented us an interesting subject with a very original presentation!
Indeed, while Noël explained the origins of browsers and their evolution through time, Pierre drew the slides in _real time_ on the stage.

.Once upon a time... the browsers
image::live-drawing.jpg[Once upon a time... the browsers, width = 800]

> Thanks to this duet, we can get hooked directly to the story that is told to us.

Reviewer: could be

```suggestion
Thanks to this duet, we can get hooked directly into the story that is told to us.
```

or 

```suggestion
Thanks to this duet, we can get directly hooked into the story that is told to us.
```

probably the former is better

> I admire a lot the people who manage to go out of the frame to present their subject, a big bravo to them!

Reviewer: ```suggestion
I really admire the people who manage to go out of the frame to present their subject, a big bravo to them!
```

For Mario Kart Deluxe 8 players, here are the main techniques I didn't know:

. If you have a piece as a main object, *keep it* because the next object will necessarily be a *protection object*
. When a red carapace comes on you, you can enter a wall at the last moment to avoid it
. To *avoid a blue carapace*, you can use a *mushroom* when the carapace falls on you

Afterwards, Yann showed us several advanced methods, but to be honest, I don't think I'll ever be able to reproduce them...

The weekend after the conference was an opportunity to test out all of his techniques and to my surprise, I really improved!
Considering my initial level, I had a lot of room for improvement...
If you're a Mario Kart player on Nintendo Switch, I highly recommend watching this talk a few times!

=== "Camping Paradigme : Organise tes vacances en mode Agile !" (Camping Paradigm: Organize your vacations in Agile mode !)

https://twitter.com/Laurie78645718[Laurie Coquillat] imagined and created a board game to explain *Agile* methods in a playful way.

.Camping Paradigm: Organize your vacations in Agile mode !
image::agile1.jpg[Camping Paradigm: Organize your vacations in Agile mode !, width = 800]

> In this workshop, you have to organize your group vacation with unforeseen events and constraints in the best possible way.

Reviewer: ```suggestion
In this workshop, you had to organize your group vacation with unforeseen events and constraints in the best possible way.
```

This parallel with an IT project allows participants to discover the world of agility in an original way.

.The game board
image::agile2.jpg[The game board, width = 800]

There are several points to take into account during our adventure such as the mode of transport used, the activities, the meals or the chosen accommodation.
During the game, you may come across some unexpected situations that you have to deal with as a group.

> Laurie is present throughout the game to guide and help us.

Reviewer: ```suggestion
Laurie was present throughout the game to guide and help us.
```

> At the end of the game, she highlights the parallels with our IT projects to explain the different Agile frameworks and their use cases.

Reviewer: ```suggestion
At the end of the game, she highlighted the parallels with our IT projects to explain the different Agile frameworks and their use cases.
```

1. a *generator* that will produce random images at the beginning and then more and more realistic as you train
2. a bank of real images
3. a *discriminator* that will have to detect if the image presented to it comes from the generator or from the image bank

The goal of the generator is to produce the most realistic image possible to deceive the discriminator and the goal of the discriminator is to detect as well as possible the false images generated.
After many training cycles, the two modules will improve and become more and more efficient.

The first trick to detect false images generated by this model comes from the *image bank used*.
Indeed, the faces coming from this group of photos have a particularity: _the eyes are always placed in the same place_.
Therefore, the AI will try to create images as close as possible to the real ones and will therefore also place the eyes in the same place.
The most naive solution is to look at the position of the eyes on the image and if it matches the known position, the image is false.
But this solution is not very robust to changes in the pictures, so the presenter developed a more advanced method.

> To do this, it has set up a service that retrieves and stores images from the site_thispersondoesnotexist_.

Author: Nice catch :D

It has made several optimizations in order not to store the raw images on the server, as this would take up too much disk space.
Then, when you want to detect a fake face, the tool will compare with its database and according to a score, send back yes or no.
If you want to test, Mathis hosts a https://fada.h25.io/[website] allowing to detect fake images directly from the browser.
Moreover, this website displays interesting information such as look-alikes, when the fake image was generated, or the eyes technique explained before.

=== "Il était une fois... les navigateurs" (Once upon a time... the browsers)

*Noël Macé* and *Pierre Tibulle* presented us an interesting subject with a very original presentation!
Indeed, while Noël explained the origins of browsers and their evolution through time, Pierre drew the slides in _real time_ on the stage.

.Once upon a time... the browsers
image::live-drawing.jpg[Once upon a time... the browsers, width = 800]

> Thanks to this duet, we can get hooked directly into the story that is told to us.

Reviewer 2: ```suggestion
Thanks to this duet, we can get hooked directly into the story that is told to us.
```
I'm not 100% sure what this means. Does this mean there were engaging speakers?

I really admire the people who manage to go out of the frame to present their subject, a big bravo to them!

By the way, https://twitter.com/ptibulle[Pierre] has posted on his twitter feed several summaries of talks he participated in (like the one about deep fakes).

=== "Mais, à quoi ça sert un DevRel ?" (But, what is a DevRel for?)

We had the privilege to receive two unicorns on stage to explain us the job of **DevRel** (Developer Relations).

.But, what is a DevRel for?
image::devrel.jpg[But, what is a DevRel for?, width = 800]

I really enjoyed this talk because the term *DevRel* was obscure to me.

> I imagined a person who goes from conference to conference to sell his company.

Reviewer 2: ```suggestion
I imagined a person who goes from conference to conference to sell their company.
```
Better to be gender agnostic here.
